### DISCUSSION TITLE
Help pick a new name for `swift_bazel`

### DISCUSSION THREAD

**Discussion by cgrindel** [cgrindel/rules_swift_package_manager#310] [Polls]
## Why change the name?

Names are hard. I picked `swift_bazel` as the repository name because I could not decide on a better one. I wanted something descriptive, but not confining. 

The problem is that it is not a great name. It has the potential to confuse new Bazel users as being the primary repository for Swift development. It is not. It is meant to augment `rules_swift` and `rules_apple` providing functionality that they do not provide. Also, it does not provide any context about what it provides.

## Why change the name now?

I am wrapping up work that adds support for [bzlmod](https://bazel.build/external/module). Before I submit an entry to the [Bazel Central Registry](https://registry.bazel.build/), I want to make sure that the name is appropriate.

## What is in this repository?

The `swift_bazel` repository was created to supplant the functionality in [rules_spm](https://github.com/cgrindel/rules_spm/). However, it also contains functionality to reduce some of the tedium working with Swift in a Bazel repository. The following is a quick recap of this repository's features:

1. `gazelle update-repos`: Resolving Swift package dependencies into Bazel repository rules.
2. `gazelle update/fix`: Generate `swift_xxx` declarations for Swift source files. The interesting part here is related to the mapping of the Bazel targets for the Swift package products.
3. `swift_package` and `local_swift_package`: Repository rules that build the package and make their product/targets available to other Bazel rules.

## If the repository name changes, will current `swift_bazel` users need to update their code?

Not right away. GitHub supports repository renaming by ensuring that the original URLs continue to work. However, once we do land on a new name, I highly recommend that folks use it in their Bazel code.

## Help me name this repository

Below is a list of some repository names that have been kicked around. Please pick your favorite or provide your own suggestion. I will then use the responses from the poll to select a new name.

**Comment by jpsim** [comment]
How about `rules_swiftpm`?

**Comment by cgrindel** [comment]
I want to thank everyone who participated in the poll. Based on the results, it looks like `rules_swift_package_manager` is the preferred repository name. I will start the rename and publication to the Bazel Central Registry.
